util.go

```diff
@@ -44,3 +45,35 @@ func (genmsg *NFGenMsg) Decode(b []byte) {
 	genmsg.Version = b[1]
 	genmsg.ResourceID = binary.BigEndian.Uint16(b[2:])
 }
+
+// GetFirstIPFromCIDR returns the first IP address from a CIDR.
+func GetFirstIPFromCIDR(cidr string) (*net.IP, error) {
+	_, subnet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return nil, err
+	}
+
+	mask := binary.BigEndian.Uint32(subnet.Mask)
+	ip := binary.BigEndian.Uint32(subnet.IP)
+
+	// find the final address
+	firstIP := make(net.IP, 4)
```

Comment:
This is IPv4 specific. Can the function also be made to work with IPv6? I know NAT is rare with IPv6, but sometimes used.